ssl/statem/statem_srvr.c

```diff
@@ -1850,7 +1881,7 @@ static int tls_early_post_process_client_hello(SSL *s, int *pal)
      * Given s->session->ciphers and SSL_get_ciphers, we must pick a cipher
      */
 
-    if (!s->hit || s->hello_retry_request) {
+    if (!s->hit || SSL_IS_TLS13(s)) {
```

Comment:
I'm having trouble seeing how this chunk relates to the stated change.  Is it unrelated cleanup?